modules/gateway/peers.go

```diff
@@ -184,62 +177,79 @@ func (g *Gateway) managedAcceptConnNewPeer(conn net.Conn, remoteVersion string)
 		return err
 	}
 
+	// Attempt to add the peer to the node list. If the add is successful and
+	// the address is a local address, mark the peer as a local peer.
+	local := false
+	err = g.managedAddUntrustedNode(remoteAddr)
+	if err != nil && remoteAddr.IsLocal() {
+		local = true
```

Comment:
I'm pretty sure that you can lie about what your remote address is. If you say that you are a local node, you get special treatment in terms of being kicked. Don't want to do that without verifying that you are reachable at the specified address.